### PR TITLE
Improved Aimbot

### PR DESCRIPTION
**Pull request by Sys0xdbg**
Fix aimbot when in vehicle
[Video](https://www.youtube.com/watch?v=8g1ZiXLJ4E0)

### CODE REVIEW

**Comment by lonelybud** [comment]
any idea on why it aims down on ped when you are aiming from inside vehicle?

File: src/backend/looped/weapons/aimbot.cpp [hunk 1]
```diff
@@ -16,75 +16,92 @@ namespace big
 		virtual void on_tick() override
 		{
 			float local_fov_change = g.weapons.aimbot.fov;
-			for (auto ped : entity::get_entities(false, true))
+			if (PED::GET_PED_CONFIG_FLAG(self::ped, 78, 0))
```
Comment (gir489returns) [inline src/backend/looped/weapons/aimbot.cpp:19]: Replace this with PLAYER::IS_PLAYER_FREE_AIMING

File: src/backend/looped/weapons/aimbot.cpp [hunk 2]
```diff
@@ -16,75 +16,92 @@ namespace big
...
+				    self::pos.z,
+				    world_position.x,
+				    world_position.y,
+				    world_position.z);
+
+				if (distance_between_target > (g.weapons.aimbot.distance * g.weapons.aimbot.distance))
+				{
+					return;
+				}
+				if (!ENTITY::HAS_ENTITY_CLEAR_LOS_TO_ENTITY(self::ped, target_entity, 17))
```
Comment (gir489returns) [inline src/backend/looped/weapons/aimbot.cpp:57]: Replace with HAS_ENTITY_CLEAR_LOS_TO_ENTITY_ADJUST_FOR_COVER.

**Review by gir489returns — CHANGES_REQUESTED** [review]
The aimbot still is in a terribly unreliable state. https://youtu.be/gFlzA4MzRO4

In my video, you can see it still tries to grab people who are behind cover or are completely unshootable. It also fails to grab targets that are "too far" if I crank the distance up to 1000, it just aims over their head.

**Comment by TheGreenBandit** [comment]
I'm

File: src/backend/looped/weapons/aimbot.cpp [hunk 3]
```diff
@@ -93,20 +110,21 @@ namespace big
...
 				//  Then it will start leading to 0,0,0 coords.Aim will start pointing at 0,0,0 as well.
 				if (aim_lock.x == 0.f && aim_lock.y == 0.f && aim_lock.z == 0.f)
 					return;
 
-				constexpr float RADPI = 180.0f / std::numbers::pi;
-				float magnitude       = std::hypot(camera_target.x, camera_target.y, camera_target.z);
-				float camera_heading  = atan2f(camera_target.x, camera_target.y) * RADPI;
+				float RADPI          = 180.0f / std::numbers::pi;
+				float camera_heading = atan2f(camera_target.x, camera_target.y) * RADPI;
+				float magnitude      = sqrtf(camera_target.x * camera_target.x + camera_target.y * camera_target.y
```
[the remark this answers is not in the document]
Comment (Sys0xdbg) [reply src/backend/looped/weapons/aimbot.cpp:122]: Yes, mistake because i used old commit.

**Comment by gir489returns** [comment]
@navmodder What is the status of your work? I haven't been able to fix the issues I mentioned.

**Comment by Sys0xdbg** [comment]
> @gir489returns 

I fixed the aiming issue, but that vehicle issue is still there. Tried adjusting the pitch for both vehicle and weapon, but no luck. Maybe we should just go back to the old way for now? My bad for pushing this broken thing.

**Comment by gir489returns** [comment]
@navmodder Please take a look at the source code posted with #2701 and see if the math there makes yours better. Also, see if you can implement his visual checks, they seem to be superior, although it looks like he's just calling the native and not waiting like the documentation says to. If you can't get it working, I will properly port his code 1:1 over.

**Comment by gir489returns** [comment]
@navmodder It has been 3 weeks. What is the status of porting the code from #2701 to your codebase?

**Comment by Yimura** [comment]
No response from author and PR went stale.

**Pull request closed without merging**